zboxcore/sdk/chunked_upload.go

```diff
@@ -527,38 +509,69 @@ func (su *ChunkedUpload) processUpload(chunkStartIndex, chunkEndIndex int, fileS
 			return err
 		}
 
-		go func(b *ChunkedUploadBlobber, buf *bytes.Buffer, form ChunkedUploadFormMetadata) {
+		go func(idx uint64, b *ChunkedUploadBlobber, buf *bytes.Buffer, form ChunkedUploadFormMetadata) {
 			err := b.sendUploadRequest(ctx, su, chunkEndIndex, isFinal, encryptedKey, buf, form)
 
 			// channel is not closed
 			if ctx.Err() == nil {
-				wait <- err
+				wait <- UploadError{
```

Comment:
We should push to wait anyway. Imagine if the ctx is canceled early and we only pushed half  of the num to wait, then the for loop below that reading err from wait channel will get stuck.

Reply:
fixed